open-api/rest-catalog-open-api.py

```diff
@@ -991,6 +991,9 @@ class ValueMap(BaseModel):
 
 class DataFile(ContentFile):
     content: str = Field(..., const=True)
+    first_row_id: Optional[int] = Field(
+        None, alias='first-row-id', description='Starting row ID to assign to new rows'
```

Comment:
nit: to be consistent with the spec, maybe the description can be `The first row ID assigned to the first row in the data file`

Reply:
I re-used the description which we have in the DataFile 

https://github.com/apache/iceberg/blob/5d2230ead79da64a8c871a02eb1304a94aaece5c/api/src/main/java/org/apache/iceberg/DataFile.java#L102

The suggested description seems reasonable to me, if you feel strongly, happy to change in all 3 places, please let me know your thoughts

Reply:
I see. I think the spec description is a little more clear to me than the description from code.
https://iceberg.apache.org/spec/#snapshots

This is for the snpshot `first-row-id` field. but the concept is the same and hence description can be similar.
```
first-row-id	The first _row_id assigned to the first row in the first data file in the first manifest, see [Row Lineage](https://iceberg.apache.org/spec/#row-lineage)
```

Reply:
sure then ! though seems like we do have first_row_id missing from the snapshot too here, should i add this as well ?

Reply:
now you mentioned it. there is also an `next-row-id` in table metadata and `first-row-id` in the manifest file.

Reply:
added in Snapshots and TableMetadata these new fields, manifest is presently not present in rest-spect we should be fine :), seems like we need to add key-id too let me add this in follow-up and keep this pr only about row-lineage